pkgs/unified_analytics/lib/src/analytics.dart

```diff
@@ -225,6 +225,15 @@ abstract class Analytics {
   /// out of the configuration file.
   Map<String, ToolInfo> get parsedTools;
 
+  /// Use this list to check for events that have been emitted when
+  /// invoking the send method.
+  ///
+  /// This list will only get populated when running the [Analytics.test]
+  /// constructor, [Analytics.development], or an instance of [FakeAnalytics].
+  /// It will otherwise remain empty when used in production.
```

Comment:
Do we need to check this in all of those instances?

To put it another way, can we implement this on some explicitly test interface that implements `Analytics`?

Reply:
👍  This was a good call, it was simple enough for me to use the `Analytics.test` constructor to return a `FakeAnalytics` instance instead of the base implementation class.